src/components/sidebar/sidebar.pw.tsx

```diff
@@ -0,0 +1,435 @@
+
+      await assertCssValueIsApproximately(
+        sidebarPreview(page),
+        "width",
+        parseInt(width)
+      );
+    });
+  });
+
+  test("check component has expected header", async ({ mount, page }) => {
+    await mount(<SidebarComponentWithHeading />);
+
+    const sidebarHeaderComponent = getComponent(page, "sidebar-header");
+
+    await expect(sidebarHeaderComponent).toBeVisible();
+
+    const sidebarPreviewHeading = sidebarPreview(page).locator("h3").nth(0);
+    await expect(sidebarPreviewHeading).toHaveText("Sidebar Header");
+  });
+
+  test("check component has role", async ({ mount, page }) => {
+    await mount(<SidebarComponent role={CHARACTERS.STANDARD} />);
+
+    await expect(sidebarPreview(page)).toHaveAttribute(
+      "role",
+      CHARACTERS.STANDARD
+    );
+  });
+
+  test("check component has expected data-element", async ({ mount, page }) => {
+    await mount(<SidebarComponent data-element={CHARACTERS.STANDARD} />);
+
+    await expect(sidebarComponent(page)).toHaveAttribute(
+      "data-element",
+      CHARACTERS.STANDARD
+    );
+  });
+
+  test("check component has expected data-role", async ({ mount, page }) => {
+    await mount(<SidebarComponent data-role={CHARACTERS.STANDARD} />);
+
+    await expect(sidebarComponent(page)).toHaveAttribute(
+      "data-role",
+      CHARACTERS.STANDARD
+    );
+  });
+
+  ([true, false] as const).forEach((boolVal) => {
+    test(`verify visibility of sidebar component when disableEscKey is ${boolVal} and Escape key is pressed`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<SidebarComponentWithOnCancel disableEscKey={boolVal} />);
+
+      const sidebarPreviewElement = sidebarPreview(page);
+      await sidebarPreviewElement.press("Escape");
+      await waitForAnimationEnd(sidebarPreviewElement);
+
+      if (boolVal) {
+        await expect(sidebarPreviewElement).toBeInViewport();
+      } else {
+        await expect(sidebarPreviewElement).not.toBeInViewport();
+      }
+    });
+  });
+
+  ([true, false] as const).forEach((boolVal) => {
+    test(`check component with open prop set to ${boolVal}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<SidebarComponent open={boolVal} />);
+
+      const sidebarPreviewElement = sidebarPreview(page);
+
+      if (boolVal) {
+        await expect(sidebarPreviewElement).toBeInViewport();
+      } else {
+        await expect(sidebarPreviewElement).not.toBeInViewport();
+      }
+    });
+  });
+
+  test("should render component with focusableContainers", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<SidebarComponentFocusable />);
+
+    const toastElement = getComponent(page, "toast");
+
+    await expect(toastElement).not.toBeVisible();
+
+    const openToastElement = getDataElementByValue(page, "open-toast");
+    await openToastElement.click();
+
+    await expect(toastElement).toBeVisible();
+
+    const toastElementChildrenChildren = page
```

Comment:
**suggestion**: lol at `toastElementChildrenChildren` as a const name - I assume that's the nonsense output by the automated conversion script I wrote 😅 But I'm sure you can come up with a better name (`toastCloseButton` maybe?)